.github/workflows/release_maven.reusable.workflow.yml

```diff
@@ -0,0 +1,266 @@
+        shell: bash
+
+      # Validate Docker tags
+      # Always use the common action for Docker tags validation. Docker image tags
+      # could have additional or different rules, and rules could change in time.
+      # This way we enforce that all docker tags pushed to the Docker registry follow
+      # same rules whatever the workflow which created them,
+      # event though Maven version provided as workflow input has already been checked,
+      # or the workflow itself computes the version
+      - name: Validate docker tags
+        uses: ./.github/symphony-devops-workflows/actions/tags_validate_clean
+        with:
+          tags: '${{ steps.compute_docker_tags.outputs.DOCKER_TAG_WITH_VERSION }}'
+          allowEmpty: 'false'
+
+      # Build and deploy project
+      - name: Maven build
+        uses: ./.github/symphony-devops-workflows/actions/maven_build_and_tag
+        with:
+          # Maven options
+          pom-xml-file: ${{ inputs.pom-xml-file }}
+          maven_goals: ${{ inputs.maven-goals }}
+          maven_other_options: -Dbuilder.version=${{ steps.compute_build_number.outputs.build-number }} ${{ inputs.maven-other-options }}
+          # Disable Git tagging (already done when updating pom.xml version)
+          git_tag: ''
+
+      # Docker
+      - name: Docker build
+        id: docker_build_tag_push
+        # Run docker build only when a docker file has been specified (not empty value)
+        if: ${{ !cancelled() && !failure() && inputs.docker-file != '' }}
+        uses: ./.github/symphony-devops-workflows/actions/docker_build_tag_push
+        with:
+          docker-file: ${{ inputs.docker-file }}
+          docker-tag-push: 'true'
+          docker-tags: '${{ steps.compute_docker_tags.outputs.DOCKER_TAG_WITH_VERSION }}'
+          docker-image-prefix: '${{ steps.compute_docker_tags.outputs.DOCKER_IMAGE_TAG_PREFIX }}'
+
+      - name: Kubernetes update deployment
+        if: ${{ !cancelled() && !failure() && inputs.kubernetes-update-deployment-descriptor && inputs.docker-file != '' && steps.docker_build_tag_push.outputs.docker-pushed == 'true' }}
+        uses: ./.github/symphony-devops-workflows/actions/kubernetes_update_deployment
+        with:
+          # Path is relative to deployer project root
+          deployment-file: '${{ inputs.symphony-microservice-name }}/deploy-service.yaml'
+          # New value for docker image in deployment file
+          docker-image: '${{ steps.compute_docker_tags.outputs.DOCKER_IMAGE_TAG_PREFIX }}:${{ steps.compute_docker_tags.outputs.DOCKER_TAG_WITH_VERSION }}'
+          fail-if-no-file-change: 'true'
+          # WARNING : relative path to current workspace, without leading "./" (for some reason this breaks replacement)
+          workspace-checkout-dir: 'symphony-kubernetes-deployer'
+          # The revision (branch) to be updated
+          symphony-kubernetes-deployer-ref: 'dev'
```

Comment:
shouldn't this value come from a user input into workflow?

Reply:
Process needs to be defined.
Human input is possible, but also can be error prone. For example, overwriting QA by mistake.
For now I would prefer having the release to be always and only deployed to Dev. Then a human could check that everything is fine on Dev, and then ask deployment to QA (or SIM…) when QA Team is ready, possibly using a separate workflow.
Let’s discuss that with the Team.